reyna/src/com/b2msolutions/reyna/services/StoreService.java

```diff
@@ -13,39 +13,51 @@
 
 	public static final String MESSAGE = "com.b2msolutions.reyna.MESSAGE";
 
+	private static Long storageSizeLimit = null;
+
 	public StoreService() {
 		super(StoreService.class.getName());
 
 		Logger.v(TAG, "StoreService()");
 	}
-	
+
 	public static void start(Context context, Message message) {
-        Logger.v(TAG, "start");
+		Logger.v(TAG, "start");
 
 		Intent service = new Intent(context, StoreService.class);
 		service.putExtra(StoreService.MESSAGE, message);
 		context.startService(service);
 	}
 
-    public static void setLogLevel(int level) {
-        Logger.v(TAG, "setLogLevel: " + level);
+	public static void setLogLevel(int level) {
+		Logger.v(TAG, "setLogLevel: " + level);
+
+		Logger.setLevel(level);
+	}
 
-        Logger.setLevel(level);
-    }
+	public static void setCellularDataBlackout(Context context, TimeRange range) {
+		Logger.v(TAG, "setCellularDataBlackout: " + range);
+		new Preferences(context).saveCellularDataBlackout(range);
+	}
+
+	public static void setStorageSizeLimit(long limit) {
```

Comment:
Can you please change this function to be setMaximumSize(long numBytes) to be equivalent to android API.
then value is stored in shared-preferences not as static, for below reasons
1- static variable make the component running even if not doing any thing
2-With devices that usually reboot and process killed to save resources, it will be better to be saved on disk

Can this function, plus saving the new value, compare it against old one and if new value is smaller then it, shrink the database,instead of shrink at the insert time.

Thanks,
Youhana